packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx

```diff
@@ -132,6 +138,24 @@ describe('ResourceModal', () => {
         'https://www.google.com/',
       );
     });
+    it('renders the delete button if onDelete function is provided', () => {
+      renderResourseModal({
+        type: 'Link',
+        externalLink: 'https://www.google.com/',
+        onDelete: jest.fn(),
+      });
+      expect(screen.getByRole('button', { name: /delete/i })).toBeVisible();
+    });
+    it('allows a resource to be deleted', async () => {
+      const onDelete = jest.fn();
+      renderResourseModal({ onDelete });
+      enterType('Link');
+      enterTitle('a title');
+      enterLink('http://example.com');
+      const deleteButton = deleteClick();
+      await waitFor(() => expect(deleteButton).toBeEnabled());
+      expect(onDelete).toBeCalledWith();
```

Comment:
If you want to check the method is called use toBeCalled.

Or do you want to check what onDelete has been called with?

Reply:
Clarified